### PR TITLE
[Qute] Add language support for qute

### LINKED ISSUE #178: [Qute] Add language support for qute
Would be great if we could add support for qute template language https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/qute-reference.adoc

Mvp would be syntax highlighting similar to how other template languages syntax highlighting works. 

Nice to have feature would be LSP support by understanding the inject and parameter declarations and then provide content assist. 

Looking into having a preview feature that would use a data yaml for it's rendering would be great. I.e. if you have a mytemplate.qt file then mytemplate.qt.yaml could be used by default to make it super trivial. 

-

### PR DESCRIPTION
**Pull request by angelozerr**
[Qute] Add language support for qute

Fixes #178

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
To play this Qute support:

 * get the PR from Quarkus LS which host the Qute LS https://github.com/redhat-developer/quarkus-ls/pull/428
 * get this PR

In vscode:

 * `npm install`
 * `npm run build-qute-server` (to create the uber Jar of Qute LS)
 * `npm run build-qute-ext` (to create the Jar of Qute JDT)

and after that you should benefit with Qute support (I hope)

**Comment by angelozerr** [comment]
Here a little demo with Qute support:

QuteDemo

To test this support I suggest that you read https://quarkus.io/guides/qute-reference (I have not supported the full features of Qute, so please create an issue for each feature which is not supported or which doesn't work as expected)/

There is too support with Quarkus / Qute https://quarkus.io/guides/qute-reference#quarkus_integration To test this support you can play with https://github.com/agoncal/agoncal-articles/tree/master/2021-06-QutePanache and read  the https://antoniogoncalves.org/2021/06/07/a-qute-way-to-visualise-data-with-panache/ article

You will need to rename qute file with `*.qute.html`. For instance you must rename https://github.com/agoncal/agoncal-articles/blob/master/2021-06-QutePanache/src/main/resources/templates/BookPage/books.html to books.qute.html

QuarkusQuteDemo

File: src/qute/languageServer/requirements.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    const config = workspace.getConfiguration();
+    return config.get<string>('java.home');
+}
+
+function checkJavaVersion(javaHome: string): Promise<number> {
+    return new Promise((resolve, reject) => {
+        cp.execFile(javaHome + '/bin/java', ['-version'], {}, (error, stdout, stderr) => {
+            const javaVersion = parseMajorVersion(stderr);
+            if (javaVersion < 11) {
+                openJDKDownload(reject, `Java 11 or more recent is required to run 'Tools for MicroProfile'. Please download and install a recent JDK.`);
```
Comment (fbricon) [inline src/qute/languageServer/requirements.ts]: run quarkus

Comment (angelozerr) [reply src/qute/languageServer/requirements.ts]: We should mention `Qute`, no?

Comment (angelozerr) [reply src/qute/languageServer/requirements.ts]: I mean `run Qute support` for instance.

Comment (fbricon) [reply src/qute/languageServer/requirements.ts]: yeah

Comment (angelozerr) [reply src/qute/languageServer/requirements.ts]: fixed

File: package.json [hunk 2]
```diff
@@ -334,13 +338,15 @@
     "@redhat-developer/vscode-redhat-telemetry": "^0.4.2",
     "ejs": "^2.7.1",
     "expand-home-dir": "0.0.3",
+    "find-java-home": "^1.0.0",
```
Comment (fbricon) [inline package.json]: try 1.2.2

Comment (angelozerr) [reply package.json]: fixed

**Comment by fbricon** [comment]
thanks @angelozerr!

**Pull request merged**